### PR TITLE
Ml calendar apis

### PR DESCRIPTION
**Pull request by codebrain**
Implementation of the `6.4.0` ML calendar APIs - see here: https://github.com/elastic/elasticsearch-net/issues/3552

Requires port to `master`

### CODE REVIEW

**Comment by Mpdreamz** [comment]
@codebrain mind retargetting this against `6.6` and investigate the CI failures? 👍

**Comment by russcam** [comment]
Changed base to 6.6

File: src/Nest/XPack/MachineLearning/DeleteCalendarJob/DeleteCalendarJobResponse.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Nest
+{
+	/// <summary>
+	/// The response from deleting a calendar job.
+	/// </summary>
+	public partial interface IDeleteCalendarJobResponse : IResponse
+	{
+		[JsonProperty("calendar_id")]
+		Id CalendarId { get; }
```
[the remark this answers is not in the document]
Comment (codebrain) [reply src/Nest/XPack/MachineLearning/DeleteCalendarJob/DeleteCalendarJobResponse.cs]: Good catch, will change to `string` - I think that is better and will likely play nicer with a different serialiser.

File: src/Nest/XPack/MachineLearning/PostCalendarEvents/ElasticClient-PostCalendarEvents.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+
+namespace Nest
+{
+	public partial interface IElasticClient
+	{
+		/// <summary>
+		/// Instantiates a machine learning calendar.
```
[the remark this answers is not in the document]
Comment (codebrain) [reply src/Nest/XPack/MachineLearning/PostCalendarEvents/ElasticClient-PostCalendarEvents.cs]: Fixed.

File: src/Nest/XPack/MachineLearning/GetCalendars/ElasticClient-GetCalendars.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+
+namespace Nest
+{
+	public partial interface IElasticClient
+	{
+		/// <summary>
+		/// Retrieves configuration for machine learning jobs.
```
[the remark this answers is not in the document]
Comment (codebrain) [reply src/Nest/XPack/MachineLearning/GetCalendars/ElasticClient-GetCalendars.cs]: Fixed.

**Comment by codebrain** [comment]
Ported to `master`

**Comment by codebrain** [comment]
Subsequent commit made to address other types in response objects, not merged as part of this commit. This was also ported to `master`:
https://github.com/elastic/elasticsearch-net/commit/06a4d3ed0e09b60b3684e6ee8ce86590a0e6a582

**Pull request merged**